packages/@react-aria/listbox/src/useOption.ts

```diff
@@ -72,12 +79,18 @@ export function useOption<T>(props: AriaOptionProps, state: ListState<T>, ref: R
   let optionProps = {
     role: 'option',
     'aria-disabled': isDisabled,
-    'aria-selected': isSelected,
-    'aria-label': props['aria-label'],
-    'aria-labelledby': labelId,
-    'aria-describedby': descriptionId
+    'aria-selected': isSelected
   };
 
+  // Safari with VoiceOver on macOS misreads options with aria-labelledby or aria-label as simply "text".
+  // We should not map slots to the label and description on Safari and instead just have VoiceOver read the textContent. 
+  // WebKit issue: https://bugs.webkit.org/show_bug.cgi?id=209279
+  if (!isSafariMacOS) {
```

Comment:
should we be looking for something other than the browser? so that when safari is fixed, things will just work correctly?

Reply:
not sure what we'd look for here unfortunately. there's no way to access the accessibility tree from JS.

Reply:
I was afraid of that, I couldn't think of anything else either :(